tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/TypeUtils.java

```diff
@@ -675,13 +671,14 @@ public static boolean isMemberType(Element element) {
 	}
 
 	public static String getGeneratedClassFullyQualifiedName(TypeElement element, String packageName, boolean jakartaDataStyle) {
-		final String packageNamePrefix = !packageName.isEmpty() ? packageName + "." : "";
-		final String className = Arrays.stream(
-						element.getQualifiedName().toString().substring( packageNamePrefix.length() ).split( "\\." ) )
-				.map( StringUtil::removeDollar )
-				.map( part -> jakartaDataStyle ? '_' + part : part + '_' )
-				.collect( Collectors.joining( "." ) );
-		return packageNamePrefix + className;
+		final StringBuilder builder = new StringBuilder( !packageName.isEmpty() ? packageName + "." : "" );
+		final int length = builder.length();
+		for ( String s : element.getQualifiedName().toString()
+				.substring( length ).split( "\\." ) ) {
```

Comment:
Our convention is to use `org.hibernate.internal.util.StringHelper.split()` instead of the regex here.

Reply:
> I am big fan of regex

Heh, I hate them with the fire of a thousand suns 😂